src/app-init/ApiManager.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  config,
+  setApi,
+  useMocks,
+  getUsername,
+  getToken,
+  loginUser,
+} from '@entando/apimanager';
+
+import { addErrors, addToast, TOAST_WARNING } from '@entando/messages';
+
+import { login } from 'api/login';
+
+class ApiManager extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      devMode: process.env.NODE_ENV === 'development',
+    };
+  }
+
+  componentDidMount() {
+    const { store } = this.props;
+    config(store);
+
+    const { dispatch, getState } = store;
+
+    dispatch(
+      setApi({
+        domain: process.env.REACT_APP_DOMAIN,
+        useMocks: process.env.REACT_APP_USE_MOCKS === 'true',
+      }),
+    );
+
+    const state = getState();
+    const mockMode = useMocks(state);
+
+    if (mockMode) this.announceMockMode();
+
+    const { devMode } = this.state;
+
+    if (devMode && !mockMode && !this.isUserLogged()) {
+      this.performAutoLogin();
+    }
+  }
+
+  isUserLogged() {
+    const { store } = this.props;
+    const { getState } = store;
+    const state = getState();
+    const username = getUsername(state);
+    const token = getToken(state);
+
+    return username && token;
+  }
+
+  announceMockMode() {
+    const { store } = this.props;
+    const { dispatch } = store;
+    dispatch(addToast('This application is using mocks', TOAST_WARNING));
+  }
+
+  performAutoLogin() {
+    const { store } = this.props;
+    const { dispatch } = store;
+    const username = 'admin';
+    const password = 'adminadmin';
+    login(username, password).then(response => response.json())
+      .then(json => dispatch(loginUser(username, json.access_token)))
+      .catch(e => dispatch(addErrors([e.message])));
+  }
+
+
+  render() {
+    const { children } = this.props;
+    return <>{children}</>;
```

Comment:
use fragments if you have to, not an empty tag.
I'm also not sure this should be a component at all.

Reply:
The empty tag is a short syntax for fragments and personally I like it: https://reactjs.org/docs/fragments.html

About the component, IMO the best option would be having an API provider component (with function props) + HOC (or hook) straight into `frontend-libraries` repo (but without autologin stuff), maybe decoupled from redux (and in a second step even from authentication, as already said in the past).

Reply:
yes, we do use it as a simplified version than using <Fragment>. We've used it around entando-pwa for quite some time now. @nicpuddu let us know if this is fine for you or not so that I'll adjust accordingly. 

For the component, I did same style with entando-pwa's ApiManager component so it will be a rather smooth transition when we can easily shift to `frontend-libraries` plan soon once we get our hands to it.

Reply:
as long as it applied project wide it is fine